lib/saluki-env/src/workload/store.rs

```diff
@@ -384,22 +400,29 @@ impl MemoryBounds for TagStore {
     }
 }
 
-/// A point-in-time snapshot of the unified
-#[derive(Debug, Default)]
-pub struct TagSnapshot {
+#[derive(Default)]
+struct TagSnapshot {
     low_cardinality_entity_tags: AHashMap<EntityId, TagSet>,
     high_cardinality_entity_tags: AHashMap<EntityId, TagSet>,
 }
 
-impl TagSnapshot {
+/// A snapshotter that supports concurrently querying the tag store.
+#[derive(Clone)]
+pub struct TagSnapshotter {
```

Comment:
Nitpicking, but I don't love this name. IIUC this is a pretty simple read handle or view, while "snapshotter" alludes to some more complex logic that I don't think exists. This made the code more confusing to read, because it seemed like there should be something to this concept of snapshotting and I wasn't seeing it.

Reply:
That's a fair point, considering I thought the same thing after renaming it.

I'll tweak it to be something closer to the intended usage.

Reply:
I think/hope this sounds more reasonable now as `TagStoreQuerier`.